pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -318,8 +326,10 @@ impl InMemoryLayer {
         trace!("initializing new empty InMemoryLayer for writing on timeline {timeline_id} at {start_lsn}");
 
         let file = EphemeralFile::create(conf, tenant_shard_id, timeline_id).await?;
+        let key = InMemoryLayerKey(file.path());
```

Comment:
If this just has to be unique, then the ephemeral file id would probably handle the same thing, and be `Copy`.

Reply:
https://github.com/neondatabase/neon/pull/7215/commits/23745c232e1fc4102a0360a2a766fbecbd5fe890